apps/condo/domains/notification/adapters/firebaseAdapter.js

```diff
@@ -0,0 +1,226 @@
+        const mixed = { ...result }
+
+        fakeNotifications.forEach(({ token }) => {
+            if (token === PUSH_FAKE_TOKEN_SUCCESS) {
+                mixed.successCount++
+                mixed.responses.push(getFakeSuccessResponse())
+            }
+            if (token === PUSH_FAKE_TOKEN_FAIL) {
+                mixed.failureCount++
+                mixed.responses.push(FAKE_ERROR_RESPONSE)
+            }
+        })
+
+        return mixed
+    }
+
+    async sendNotification ({ notification, tokensData, data } = {}) {
+        console.log('sendNotification:', { notification, tokensData, data })
+
+        // If we were unable to initialize firebase, then we will always fail to deliver pushes
+        if (isNull(this.app)) return [false, { error: 'No FIREBASE_CONFIG_JSON provided or invalid' }]
+        if (!tokensData || isEmpty(tokensData)) return null
+
+        const [notifications, fakeNotifications] = this.prepareNotifications(notification, data, tokensData)
+        let result
+
+        console.log('sendNotification notifications:', notifications)
+
+        if (isEmpty(notifications)) {
+            result = this.injectFakeResults(EMPTY_RESULT, fakeNotifications)
+        } else {
+            result = await this.app.messaging().sendAll(notifications)
+                .then((result) => {
+                    const extraResult = this.injectFakeResults(result, fakeNotifications)
+
+                    return extraResult
+                })
+                .catch((error) => {
+                    return { state: 'error', error }
+                })
+        }
+
+        const isOk = !isEmpty(result) && result.successCount > 0
+
+        return [isOk, result]
+    }
+
+    // async sendData (args = {}) {
+    //     const { pushToken, pushTokens, data } = args
+    //     if (!data || isEmpty(data)) throw new Error('No notification')
+    //
+    //     const payload = this.preparePayload({ data, pushToken, pushTokens })
+    //
+    //     return this.app.messaging().send({ data, token: pushToken })
+    //         .then((messageId) => {
+    //             // Response is a message ID string.
+    //             return { state: 'success', messageId }
+    //         })
+    //         .catch((error) => {
+    //             return { state: 'error', error }
+    //         })
+    // }
+
+}
+
+module.exports = {
+    FirebaseAdapter,
+}
+
+
+// // Create a list containing up to 500 messages.
+// const messages = [];
+// messages.push({
+//     notification: { title: 'Price drop', body: '5% off all electronics' },
+//     token: registrationToken,
+// });
+// messages.push({
+//     notification: { title: 'Price drop', body: '2% off all books' },
+//     topic: 'readers-club',
+// });
+//
+// getMessaging().sendAll(messages)
+//     .then((response) => {
+//         console.log(response.successCount + ' messages were sent successfully');
+//     });
+
+
+
+// const topicName = 'industry-tech';
```

Comment:
A purpose of commented code is unclear.
If it is WIP, annotate it please or remove it until it will be ready.

Reply:
removed